base/src/com/thoughtworks/go/util/FilenameUtil.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.util;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.StringUtils;
+
+import java.io.File;
+import java.io.IOException;
+
+public class FilenameUtil {
+    public static boolean isSubdirectoryOf(String parent, String subdirectory) throws IOException {
+        final String normalizedParentPath = FilenameUtils.normalize(parent + File.separator);
+        final String normalizedSubDirPath = FilenameUtils.normalize(subdirectory + File.separator);
+        return StringUtils.isNotBlank(normalizedParentPath) && StringUtils.isNotBlank(normalizedSubDirPath) && normalizedSubDirPath.startsWith(normalizedParentPath);
+    }
+    public static boolean isFolderInsideSandbox(String path) {
```

Comment:
Can we instead rename this to `isNormalizedPathOutsideWorkingDir` or something similar to indicate that this fundamentally performs path normalization. `insideSandbox` seems to invoke the feeling that it actually does resolution of file paths.

Reply:
cool